src/Microsoft.Diagnostics.Monitoring.WebApi/Stacks/JsonStacksFormatter.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Monitoring.WebApi.Stacks
+{
+    internal sealed class JsonStacksFormatter : StacksFormatter
+    {
+        public JsonStacksFormatter(Stream outputStream) : base(outputStream)
+        {
+        }
+
+        public override async Task FormatStack(StackResult stackResult, CancellationToken token)
+        {
+            Models.StackResult result = new Models.StackResult();
+            NameCache cache = stackResult.NameCache;
+
+            foreach(Stack stack in stackResult.Stacks)
+            {
+                Models.Stack stackModel = new Models.Stack();
+                stackModel.ThreadId = stack.ThreadId;
+
+                foreach(StackFrame frame in stack.Frames)
+                {
+                    Models.StackFrame frameModel = new Models.StackFrame()
+                    {
+                        ClassName = UnknownClass,
+                        MethodName = UnknownFunction,
+                        Offset = frame.Offset,
+                        ModuleName = UnknownModule
+                    };
+                    if (cache.FunctionData.TryGetValue(frame.FunctionId, out FunctionData functionData))
+                    {
+                        frameModel.ModuleName = GetModuleName(cache, functionData.ModuleId);
+                        frameModel.MethodName = functionData.Name;
+
+                        StringBuilder builder = new StringBuilder();
```

Comment:
nit: This is going to allocate a new one for every stack frame in the process. Could you move this outside of the loop and then clear it on each iteration? Might reduce reallocations within the StringBuilder.